src/coreclr/jit/target.h

```diff
@@ -208,8 +208,156 @@ enum _regMask_enum : unsigned
 // In any case, we believe that is OK to freely cast between these types; no information will
 // be lost.
 
-#if defined(TARGET_AMD64) || defined(TARGET_ARMARCH) || defined(TARGET_LOONGARCH64) || defined(TARGET_RISCV64)
+typedef _regNumber_enum regNumber;
+typedef unsigned char   regNumberSmall;
+
+#if defined(TARGET_AMD64) || defined(TARGET_ARM) || defined(TARGET_LOONGARCH64) || defined(TARGET_RISCV64)
 typedef unsigned __int64 regMaskTP;
+#elif defined(TARGET_ARM64)
+typedef unsigned __int64 regMaskSmall;
+struct regMaskTP
+{
+    unsigned __int64 low;
+
+    regMaskTP(unsigned __int64 regMask)
+        : low(regMask)
+    {
+    }
+
+    regMaskTP()
+    {
+    }
+
+    FORCEINLINE explicit operator bool() const
+    {
+        return low != 0;
+    }
+
+    FORCEINLINE explicit operator regMaskSmall() const
+    {
+        return (regMaskSmall)low;
+    }
+
+    FORCEINLINE explicit operator unsigned int() const
+    {
+        return (unsigned int)low;
+    }
```

Comment:
This one we ought to remove as well.

Reply:
It is currently used only at one place. Ideally, all the `*refRegs` should be just gpr register set of size 4 bytes, but do not want to mix that change in this PR.

```c++
    if (ig->igFlags & IGF_BYREF_REGS)
    {
        // Record the byref regs in front the of the instructions

        *castto(id, unsigned*)++ = (unsigned)emitInitByrefRegs;
    }
```